[Feat] [snt9|2_5] Added a case: MobileNetV3-based Pulmonary Tuberculosis Detection

keepcold-yu opened the pull request:
## Description of Changes
Fixed the bugs in the demo and added the "Normal.png" file to the "images" folder.

## Test Report
### Test Environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1
### Test Result
Accuracy: 86.29%
Inference Time: 514.21 ms per image
Training Time:  8 minutes
Total Script Runtime: 12 minutes

## Related Issues
Reference any related issues or tickets. 
Fixes #4 

## Additional Notes
Include any extra information pertinent to the PR.

hongziqi commented:
Thank you for your contribution. Please update the PR title and description in accordance with the [xihe-cases contributing guidelines](https://github.com/opensourcedot/xihe-cases/blob/main/CONTRIBUTING.md).
Recommended PR title like this:
[Feat] [snt9|2_5] Added a case: MobileNetV3-based Pulmonary Tuberculosis Detection

hongziqi commented:
To enhance your (PR) description, please include the following sections:

## Description of Changes
Please provide a comprehensive overview of the modifications introduced in this PR and the motivation behind them. 

## Test Report
Provide a detailed account of the testing conducted on the [xihe.mindspore.cn](https://xihe.mindspore.cn/) platform, including environment, and results. For instance:
### Test Environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1
### Test Result
Accuracy: xxx%
Inference Time: xxx minutes per image
Training Time: xxx hours xxx minutes
Total Script Runtime: xxx hours xxx minutes
脚本总运行时间

## Related Issues
Reference any related issues or tickets. If there are none, you may omit this section

## Additional Notes
Include any extra information pertinent to the PR.

hongziqi commented:
To improve clarity and maintain consistency, we recommend updating your recent commit message .
You can amend recent commit message using the following command:
 `git commit --amend -m "[Feat] [snt9|2_5] Added a case: MobileNetV3-based Pulmonary Tuberculosis Detection"`

hongziqi commented on platform/ascend/snt9/2_5/应用实践/计算机视觉/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,2123 @@
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA8IAAAFrCAIAAACVBaf1AAAgAElEQVR4Aeydh1sTWdfAv39j0islIbRQAqHXQOjSQQUUpSmgiKJSFDXBoGJBwd5QUZrdVey6dgV0sSAWVBZdXkFFFDDk2zA6O4ZiQFAgJ4/P+9yZ3LmZ+eXumx9zz5zzfwp4AQEgAASAABAAAkAACAABIDBIAv83yP7QHQgAASAABIAAEAACQAAIAAEFaDRMAiAABIAAEAACQAAIAAEgMGgCoNGDRgYHAAEgAASAABAAAkAACAAB0GiYA0AACAABIAAEgAAQAAJAYNAEQKMHjQwOAAJAAAgAASAABIAAEAACoNEwB4AAEAACQAAIAAEgAASAwKAJgEYP ...
+    },
+    "a6b695a7-d1fc-48a8-b59a-d1d0b83ff44e.jpg": {
+     "image/jpeg": "iVBORw0KGgoAAAANSUhEUgAAA7wAAAD7CAYAAABJ9ph+AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAALsDSURBVHhe7b2FtxzF9r/9/hu/C3xxvTgXC0EDJECEuBAl7i43RghJiIe4u7srcfeEGHF3iId4br15arpmavr0nDNzdHpmP2v1Oqe6x3pPT1d9qrb8f0oQBEEQBEEQBEEQEhARvIIgCIIgCIIgCEJCIoJXEARBEARBEARBSEhE8AqCIAiCIAiCIAgJiQheQRAEQRAEQRAEISERwSsIgiAIgiAIgiAkJCJ4BUEQBEEQBEEQhIREBK8gCIIgCIIgCIKQkIjgFQR ...
```
remove empty cell

keepcold-yu commented on platform/ascend/snt9/2_5/应用实践/计算机视觉/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,2123 @@
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA8IAAAFrCAIAAACVBaf1AAAgAElEQVR4Aeydh1sTWdfAv39j0islIbRQAqHXQOjSQQUUpSmgiKJSFDXBoGJBwd5QUZrdVey6dgV0sSAWVBZdXkFFFDDk2zA6O4ZiQFAgJ4/P+9yZ3LmZ+eXumx9zz5zzfwp4AQEgAASAABAAAkAACAABIDBIAv83yP7QHQgAASAABIAAEAACQAAIAAEFaDRMAiAABIAAEAACQAAIAAEgMGgCoNGDRgYHAAEgAASAABAAAkAACAAB0GiYA0AACAABIAAEgAAQAAJAYNAEQKMHjQwOAAJAAAgAASAABIAAEAACoNEwB4AAEAACQAAIAAEgAASAwKAJgEYP ...
+    },
+    "a6b695a7-d1fc-48a8-b59a-d1d0b83ff44e.jpg": {
+     "image/jpeg": "iVBORw0KGgoAAAANSUhEUgAAA7wAAAD7CAYAAABJ9ph+AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAALsDSURBVHhe7b2FtxzF9r/9/hu/C3xxvTgXC0EDJECEuBAl7i43RghJiIe4u7srcfeEGHF3iId4br15arpmavr0nDNzdHpmP2v1Oqe6x3pPT1d9qrb8f0oQBEEQBEEQBEEQEhARvIIgCIIgCIIgCEJCIoJXEARBEARBEARBSEhE8AqCIAiCIAiCIAgJiQheQRAEQRAEQRAEISERwSsIgiAIgiAIgiAkJCJ4BUEQBEEQBEEQhIREBK8gCIIgCIIgCIKQkIjgFQR ...
```
好的 我按照要求修改完后重新提交一下pr

hongziqi commented:
Please read [Developer Contribution Hub](https://github.com/opensourcedot/xihe-cases/blob/main/dev/README.md), mv your case to `dev/ascend/snt9/2_5/`

keepcold-yu commented:
To enhance your (PR) description, please include the following sections:

## Description of Changes
Fixed the bugs in the demo and added the "Normal.png" file to the "images" folder.

## Test Report
### Test Environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1
### Test Result
Accuracy: 86.29%
Inference Time: 514.21 ms per image
Training Time:  8 minutes
Total Script Runtime: 12 minutes

## Related Issues
Reference any related issues or tickets. 
Fixes #4 

## Additional Notes
Include any extra information pertinent to the PR.

_Originally posted by @hongziqi in https://github.com/opensourcedot/xihe-cases/issues/3#issuecomment-2696417149_

hongziqi commented:
> ```
>           To enhance your (PR) description, please include the following sections:
> ```
> 
> ## Description of Changes
> Fixed the bugs in the demo and added the "Normal.png" file to the "images" folder.
> 
> ## Test Report
> ### Test Environment
> Hardware: 1*ascend-snt9|ARM: 24核 96GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> ### Test Result
> Accuracy: 86.29% Inference Time: 514.21 ms per image Training Time: 8 minutes Total Script Runtime: 12 minutes
> 
> ## Related Issues
> Reference any related issues or tickets. Fixes #4
> 
> ## Additional Notes
> Include any extra information pertinent to the PR.
> 
> _Originally posted by @hongziqi in [#3 (comment)](https://github.com/opensourcedot/xihe-cases/pull/3#issuecomment-2696417149)_

Nice! Please incorporating this content into PR description

hongziqi commented on platform/ascend/snt9/2_5/应用实践/计算机视觉/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1948 @@
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA8IAAAFrCAIAAACVBaf1AAAgAElEQVR4Aeydh1sTWdfAv39j0islIbRQAqHXQOjSQQUUpSmgiKJSFDXBoGJBwd5QUZrdVey6dgV0sSAWVBZdXkFFFDDk2zA6O4ZiQFAgJ4/P+9yZ3LmZ+eXumx9zz5zzfwp4AQEgAASAABAAAkAACAABIDBIAv83yP7QHQgAASAABIAAEAACQAAIAAEFaDRMAiAABIAAEAACQAAIAAEgMGgCoNGDRgYHAAEgAASAABAAAkAACAAB0GiYA0AACAABIAAEgAAQAAJAYNAEQKMHjQwOAAJAAAgAASAABIAAEAACoNEwB4AAEAACQAAIAAEgAASAwKAJgEYP ...
+    },
+    "a6b695a7-d1fc-48a8-b59a-d1d0b83ff44e.jpg": {
+     "image/jpeg": "iVBORw0KGgoAAAANSUhEUgAAA7wAAAD7CAYAAABJ9ph+AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAALsDSURBVHhe7b2FtxzF9r/9/hu/C3xxvTgXC0EDJECEuBAl7i43RghJiIe4u7srcfeEGHF3iId4br15arpmavr0nDNzdHpmP2v1Oqe6x3pPT1d9qrb8f0oQBEEQBEEQBEEQEhARvIIgCIIgCIIgCEJCIoJXEARBEARBEARBSEhE8AqCIAiCIAiCIAgJiQheQRAEQRAEQRAEISERwSsIgiAIgiAIgiAkJCJ4BUEQBEEQBEEQhIREBK8gCIIgCIIgCIKQkIjgFQR ...
```
has some trouble show comment quote，maybe you should review your case in [PR3](https://github.com/opensourcedot/xihe-cases/pull/3/files), and click `Load diff` to check your case
<img width="558" alt="image" src="https://github.com/user-attachments/assets/2b38e2f6-abf4-4f54-bd53-de3e2a59a0c6" />

keepcold-yu replied:
OK. I will make the revisions and upload it again.

hongziqi replied:
All right! Please read [Developer Contribution Hub](https://github.com/opensourcedot/xihe-cases/blob/main/dev/README.md), mv your case to `dev/ascend/snt9/2_5/`

keepcold-yu commented:
@hongziqi

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1911 @@
+    "\n",
+    "Excitation 操作通过两层全连接网络生成通道的权重 $ s \\in \\mathbb{R}^C $，用于重新校准每个通道的重要性。\n",
+    "\n",
+    "### 第一层全连接（降维）\n",
```
```suggestion
    "**第一层全连接（降维）**\n",
```

keepcold-yu commented:
@hongziqi

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "bbe5a1b0-6166-4deb-90e7-bf29e6341fd3",
+   "metadata": {},
+   "source": [
+    "## 1.安装指定版本的MindSpore："
```
```suggestion
    "## 环境准备\n"
    "1.安装指定版本的MindSpore："
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "a9ea4685-d75e-4382-84a2-cd54ee4ca662",
+   "metadata": {},
+   "source": [
+    "## 2.验证已安装的MindSpore版本：\n"
```
```suggestion
    "2.验证已安装的MindSpore版本：\n"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "346f5af7-5e81-4467-929d-d9938f122009",
+   "metadata": {},
+   "source": [
+    "## 3.安装kagglehub库与seaborn库\n"
```
```suggestion
    "3.安装kagglehub库与seaborn库\n"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    }
+   ],
+   "source": [
+    "!pip install kagglehub\n",
```
please add %%capture captured_output to the first line of the code block to reduce unnecessary installation information output:
```suggestion
    "%%capture captured_output\n",
    "!pip install kagglehub\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "39990b3b-f50a-4ff0-b231-e019b5866eb2",
+   "metadata": {},
+   "source": [
+    "## 4.数据集下载与存放"
```
```suggestion
    "4.数据集下载与存放"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "39e5cca5-0541-4d83-ba0b-b85ce77b9eee",
+   "metadata": {},
+   "source": [
+    "## 下载解压完成后存放格式如下\n",
```
```suggestion
    "下载解压完成后存放格式如下\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "c8ff5f75-1274-49cc-8ece-3c817de2d1a3",
+   "metadata": {},
+   "source": [
+    "# 一.激活函数构建\n",
```
```suggestion
    "## 一.激活函数构建\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "c90cef0a-1961-4825-a294-46bfbcc257a4",
+   "metadata": {},
+   "source": [
+    "# 二.SE 模块\n",
```
```suggestion
    "## 二.SE 模块\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "9774b3de-bd55-4d30-b0e6-82aaf5a9b019",
+   "metadata": {},
+   "source": [
+    "# 三.MobileNetV3 中的倒残差结构（Inverted Residual Block）\n",
```
```suggestion
    "## 三.MobileNetV3 中的倒残差结构（Inverted Residual Block）\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "71575c7f-8f6a-4a2f-9045-37c5c7fecd79",
+   "metadata": {},
+   "source": [
+    "# 四.MobileNetV3 主干网络\n",
```
```suggestion
    "## 四.MobileNetV3 主干网络\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "MobileNetV3 是一种轻量级卷积神经网络，专为移动设备设计，具有高效性和高性能的特点。其主干网络由多个 **倒残差块（Inverted Residual Blocks）** 和其他优化模块组成。\n",
+    "\n",
+    "\n",
+    "## MobileNetV3 主干网络的结构\n",
```
```suggestion
    "### MobileNetV3 主干网络的结构\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "| Global Average Pooling   | GAP                              | 1 x 1                | 960           |\n",
+    "| Fully Connected          | FC, Softmax                      | Classification Output| #Classes      |\n",
+    "\n",
+    "---\n",
```
```suggestion
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "\n",
+    "---\n",
+    "\n",
+    "## MobileNetV3-Small 和 MobileNetV3-Large 的区别\n",
```
```suggestion
    "### MobileNetV3-Small 和 MobileNetV3-Large 的区别\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "| 参数量                   | 约 2.9M                          | 约 5.4M               |\n",
+    "| 计算量（FLOPs）          | 约 0.055B                        | 约 0.217B             |\n",
+    "\n",
+    "---\n"
```
```suggestion
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "0961f4cd-2e03-400b-84d1-f98cf07851c6",
+   "metadata": {},
+   "source": [
+    "# 五、网络测试\n",
```
```suggestion
    "## 五、网络测试\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "metadata": {},
+   "source": [
+    "# 五、网络测试\n",
+    "### 打印出我们的网络结构并且创建一个张量对网络进行测试\n",
```
```suggestion
    "* 打印出我们的网络结构并且创建一个张量对网络进行测试\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "source": [
+    "# 五、网络测试\n",
+    "### 打印出我们的网络结构并且创建一个张量对网络进行测试\n",
+    "### 使用numpy的random.randn函数生成一个形状为(1, 3, 224, 224)的随机数组，数组中的元素服从标准正态分布。\n",
```
```suggestion
    "* 使用numpy的random.randn函数生成一个形状为(1, 3, 224, 224)的随机数组，数组中的元素服从标准正态分布。\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "# 五、网络测试\n",
+    "### 打印出我们的网络结构并且创建一个张量对网络进行测试\n",
+    "### 使用numpy的random.randn函数生成一个形状为(1, 3, 224, 224)的随机数组，数组中的元素服从标准正态分布。\n",
+    "### 这里的(1, 3, 224, 224)表示一个批量大小为 1、通道数为 3、高度和宽度均为 224 的图像数据。\n",
```
```suggestion
    "* 这里的(1, 3, 224, 224)表示一个批量大小为 1、通道数为 3、高度和宽度均为 224 的图像数据。\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    "### 打印出我们的网络结构并且创建一个张量对网络进行测试\n",
+    "### 使用numpy的random.randn函数生成一个形状为(1, 3, 224, 224)的随机数组，数组中的元素服从标准正态分布。\n",
+    "### 这里的(1, 3, 224, 224)表示一个批量大小为 1、通道数为 3、高度和宽度均为 224 的图像数据。\n",
+    "### 通过测试查看我们的网络结构是否正确，参数是否正确。"
```
```suggestion
    "* 通过测试查看我们的网络结构是否正确，参数是否正确。"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "7d3a83e0-439f-4a42-aa26-275263605f98",
+   "metadata": {},
+   "source": [
+    "# 六、数据划分\n",
```
```suggestion
    "## 六、数据划分\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "metadata": {},
+   "source": [
+    "# 六、数据划分\n",
+    "## 划分数据集，将原本的数据集格式  \n",
```
```suggestion
    "* 划分数据集，将原本的数据集格式：\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+    ".dataset/  \n",
+    "├── Normal/  \n",
+    "└── Tuberculosis/  \n",
+    "## 划分为  \n",
```
```suggestion
    "* 划分为：\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "2b657529-46cc-442f-929c-0038fef78cd5",
+   "metadata": {},
+   "source": [
+    "# 七、数据预处理与数据载入\n",
```
```suggestion
    "## 七、数据预处理与数据载入\n",
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "27224b27-545f-42a8-a86c-6fb1f411528e",
+   "metadata": {},
+   "source": [
+    "# 八、绘制模型评估图\n"
```
```suggestion
    "## 八、绘制模型评估图\n"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "7ed0ca44-9b55-44af-9879-372ea8e0abe5",
+   "metadata": {},
+   "source": [
+    "# 九、构建训练函数"
```
```suggestion
    "## 九、构建训练函数"
```

hongziqi commented on dev/ascend/snt9/2_5/MobileNetV3肺结核检测.ipynb:
```diff
@@ -0,0 +1,1312 @@
+   "id": "0d0a2489-f738-4eb2-8dfe-c7a15d7e54cf",
+   "metadata": {},
+   "source": [
+    "# 十、预测\n",
```
```suggestion
    "## 十、预测\n",
```

hongziqi commented:
please remove empty cell :
![image](https://github.com/user-attachments/assets/1f568a79-533a-423b-b419-45661dd5089b)

keepcold-yu commented:
@hongziqi

hongziqi commented:
please update here:
![image](https://github.com/user-attachments/assets/3e1dab47-c4cf-4a46-a1f9-05712b2f6d85)

keepcold-yu commented:
@hongziqi
